### DISCUSSION TITLE
[Bug]: Strange Build pane coloring problem

### DISCUSSION THREAD

**Discussion by iHardRock** [eranif/codelite#3515] [Setup and configuration issues] [answered]
### What happened?

There are two computers running Ubuntu (both 24.04 but it really doesn't mater) and CodeLite 17.12.0 from official monthly build (build also doesn't matter).
The only difference that first computer was installed years ago with Ubuntu ~16.04 and then was  updated to the last one as years passed by. Second computer was updated from 22.04.
Both running Unity Desktop.

On the first computer, CodeLite ALWAYS highlights errors in Build pane and clickable to navigate to line. All the time, with any CodeLite version.
On the second computer, CodeLite NEVER highlights errors an not clickable. Since first installation. Any CodeLite version.

I have tried to compile different versions of wxWidgets on second computer, and then compile CodeLite, but with no luck.
Have no idea what may lead to this difference in behavior.

Good:
![good](https://github.com/user-attachments/assets/2b59a8bb-c8c6-44cf-b271-dd88dbc31b59)

Bad:
![bad](https://github.com/user-attachments/assets/ca7ed2fc-2c78-4ce3-8ac6-8f15fc7d7cc5)

Any suggestions how to investigate?

### Version

Other

### Operating system

Linux

**Comment by eranif** [comment, marked as answer]
You failed to mention which workspace type you are using. Can you post a screenshot of the left tree view as well?

**Comment by iHardRock** [comment]
FileSystem workspace. 
Same workspace on both computers looks different in Build pane.
![screenshot](https://github.com/user-attachments/assets/5dcfbdfd-a7ef-443d-ab29-7ccb59dbf463)

**Comment by eranif** [comment]
Click on the "Wrench" icon on the right tree view toolbar (next to the orange "refresh" button)
Then select the "Build" tab and make sure that the you have a proper tool chain selected

Build again and it should be highlighted + clickable again

**Comment by iHardRock** [comment]
Solved. Thank you!
No toolchain was selected on this computer. Line was empty. Really don't know why. After setting "gnu g++" build log highlighting start working.

I suggest to show warning notification somehow for user in case of no toolchain selected. It may be not a message box, but may be a warning text inside the Build log pane.

**Comment by iHardRock** [comment]
Interesting that toolchain missed in all build configurations in my workspace on that computer, but present in same workspace on another (older) computer. Workspace file synchronized with git.

Ready to create PR.
![img](https://github.com/user-attachments/assets/e1dbd955-5960-419a-a399-c7b3acf0b2c1)

Code:
![fix](https://github.com/user-attachments/assets/2ef5d759-a23d-42d2-9485-834294cd84c8)

**Comment by eranif** [comment]
Nice, a pull request is always welcomed!

**Reply by iHardRock** [reply]
PR: #3519
